lightningd/plugin.c

```diff
@@ -943,17 +951,31 @@ void plugins_init(struct plugins *plugins, const char *dev_plugin_debug)
 		}
 		tal_free(cmd);
 	}
+}
+
+void plugins_init(struct plugins *plugins, const char *dev_plugin_debug)
+{
+	plugins->pending_manifests = 0;
+	uintmap_init(&plugins->pending_requests);
+
+	plugins_add_default_dir(plugins,
+				path_join(tmpctx, plugins->ld->config_dir, "plugins"));
+
+	/* Spawn the plugin processes before entering the io_loop */
+	plugins_start(plugins, dev_plugin_debug);
 
 	if (plugins->pending_manifests > 0)
 		io_loop_with_timers(plugins->ld);
+
+	setenv("LIGHTNINGD_PLUGIN", "1", 1);
```

Comment:
You need to move this line back to before the io_loop, BTW.

Reply:
This was because of the above, to stop `io_break`ing after this loop.